webpack/components/extensions/RegistrationCommands/fields/ActivationKeys.js

```diff
@@ -41,7 +41,12 @@ const ActivationKeys = ({
   useEffect(() => {
     handleInvalidField('Activation Keys', akHasValidValue(hostGroupId, pluginValues?.activationKeys, hostGroupActivationKeys));
   }, [handleInvalidField, hostGroupId, hostGroupActivationKeys, pluginValues]);
-
+  
+  useEffect(() => {
+    if(activationKeys.length==1){
+      onChange({ activationKeys: [activationKeys[0]?.name]})
+    }
+  }, [activationKeys]);
```

Comment:
we should avoid objects/arrays as dependencies: https://dmitripavlutin.com/react-useeffect-infinite-loop/#21-avoid-objects-as-dependencies

Reply:
Thanks @Ron-Lavi